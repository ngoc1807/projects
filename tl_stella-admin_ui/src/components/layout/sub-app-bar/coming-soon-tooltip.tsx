import { FC, ReactElement } from "react";
import { Tooltip } from "@mui/material";

interface Props {
  children: ReactElement;
  key?: string;
}

export const ComingSoonTooltip: FC<Props> = ({ children, key }) => {
  return (
    <Tooltip title="Coming Soon!" placement="top" key={key} arrow>
      {children}
    </Tooltip>
  );
};
