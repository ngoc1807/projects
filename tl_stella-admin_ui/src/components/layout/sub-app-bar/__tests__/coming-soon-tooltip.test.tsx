import { ListItemButton, ListItemText } from "@mui/material";
import { cleanup, render } from "@testing-library/react";
import { ComingSoonTooltip } from "../coming-soon-tooltip";

describe("coming-soon-tooltip", () => {
  afterEach(() => {
    cleanup();
  });
  it("should render correctly", () => {
    const { container } = render(
      <ComingSoonTooltip>
        <div>
          <ListItemButton
            selected={false}
            onClick={() => jest.fn()}
            disabled
            sx={{ cursor: "default" }}
          >
            <ListItemText primary="Receptionist" />
          </ListItemButton>
        </div>
      </ComingSoonTooltip>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
