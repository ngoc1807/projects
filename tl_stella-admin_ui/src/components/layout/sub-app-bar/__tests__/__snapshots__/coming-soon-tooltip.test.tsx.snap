// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coming-soon-tooltip should render correctly 1`] = `
<div
  aria-label="Coming Soon!"
  class=""
  data-mui-internal-clone-element="true"
>
  <div
    aria-disabled="true"
    class="MuiListItemButton-root MuiListItemButton-gutters Mui-disabled MuiButtonBase-root Mui-disabled css-1tk19ni-MuiButtonBase-root-MuiListItemButton-root"
    role="button"
    tabindex="-1"
  >
    <div
      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Receptionist
      </span>
    </div>
  </div>
</div>
`;
